src/System.Composition.Convention/src/System/Composition/Convention/PartConventionBuilder.cs

```diff
@@ -0,0 +1,822 @@
+﻿// Copyright (c) Microsoft. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+using System;
+using System.Collections;
+using System.Collections.Generic;
+using System.Composition;
+using System.Linq;
+using System.Reflection;
+using System.Text;
+
+using Microsoft.Composition.Diagnostics;
+using Microsoft.Internal;
+
+namespace System.Composition.Convention
+{
+    /// <summary>
+    /// Configures a type as a MEF part.
+    /// </summary>
+    public class PartConventionBuilder
+    {
+        private readonly Type[] _emptyTypeArray = new Type[0];
```

Comment:
@stephentoub This library targets platforms like .NET 4.5, Windows Store 8, and Windows Phone 8 which I don't think have support for `Array.Empty<T>`.